mbed_targets/mbed_targets.py

```diff
@@ -30,101 +30,46 @@
 TargetDatabaseQuery = Dict[str, TargetDatabaseQueryValue]
 
 
-@functools.total_ordering
+@dataclass(frozen=True, order=True)
 class MbedTarget:
     """Representation of an Mbed Target."""
 
-    def __init__(self, target_database_entry: dict):
-        """Create a new instance of a target.
+    board_type: str
+    board_name: str
+    product_code: str
+    target_type: str
+    slug: str
+    build_variant: Tuple[Any, ...]
+    mbed_os_support: Tuple[Any, ...]
+    mbed_enabled: Tuple[Any, ...]
+    target_entry: dict = field(default_factory=dict, repr=False, compare=False)
```

Comment:
What's the reason for having `target_entry` as one of the fields? Can it not be discarded after we've created the instance?

Reply:
It's so we can serialise the target back to JSON. I thought about trying to manage the raw entries in `MbedTargets` but I think that would get messy as we'd have to change the initialiser and hack the `Set._from_iterable` classmethod.

Reply:
If you have any ideas for decoupling the serialisation from this object do tell. I dislike this bit of code too.

Reply:
How about `to_target_entry` which mirrors `from_target_entry`. I do realize that we'd lose some information when converting from from objects returned by the api (like `id`, etc), but since we don't use this information, do we really care? Obviously, changing how this behaves would require us to re-sync the db. :)